code/modules/projectiles/guns/flamer/flamer.dm

```diff
@@ -443,6 +445,11 @@ GLOBAL_LIST_EMPTY(flamer_particles)
 	firelevel = R.durationfire + fuel_pressure*R.durationmod
 	burnlevel = R.intensityfire
 
+	//are we in weather??
+	var/area/A = get_area(src)
+	if(SSweather.is_weather_event && locate(A) in SSweather.weather_areas)
```

Comment:
you don't need the `locate` here, can directly use `(A in SSweather.weather_areas)` 

but mind it might not work properly as-is because your area could be a subarea

Reply:
Doing this or using `A.masters` completely broke functionality, couldnt tell you why but it did